entity-service-impl/src/main/java/org/hypertrace/entity/service/util/DocStoreConverter.java

```diff
@@ -89,8 +93,18 @@ private static Filter transform(AttributeFilter filter) {
       f.setFieldName(filter.getName());
       f.setOp(transform(filter.getOperator()));
       if (filter.hasAttributeValue()) {
-        transform(filter.getAttributeValue(), f, isPartOfAttributeMap(filter.getName()));
+        //transform(filter.getAttributeValue(), f);
+        if (ATTRIBUTES_LABELS_FIELD_NAME.equals(f.getFieldName()) && f.getOp() == Op.EQ) {
```

Comment:
So this would eventually go away right? Once entity service is attribute-aware and can map the LHS to an attribute type? If yes, might be worth moving it into its own function rather than inlining it.

Reply:
How does this logic interface with the logic on line 193 for a value list? Is that logic based on the RHS rather than LHS?

Reply:
> So this would eventually go away right? Once entity service is attribute-aware and can map the LHS to an attribute type? If yes, might be worth moving it into its own function rather than inlining it.

Yes. Will go away. Can be refactored to be cleaner.

> How does this logic interface with the logic on line 193 for a value list? Is that logic based on the RHS rather than LHS?
 It doesn't. Since we need the full JSON object and not just the string or whichever primitive value is wrapped in the proto, we need different logic to create the RHS for labels.